lib/svc/method/net-virtual

```diff
@@ -0,0 +1,563 @@
+        fi
+
+        # Create the overlay if necessary (the return code of dladm show-overlay may be misleading)
+        if dladm show-overlay "${link}" 2>&1 | grep "object not found" > /dev/null; then
+            eval "overlay_rule=\${SYSINFO_OverlayRule_${overlay_rule_name}}"
+
+            if [[ -z "${overlay_rule}" ]]; then
+                echo "ERROR: Missing overlay rule \"${overlay_rule_name}\" for VNIC: ${iface}"
+                return 1
+            fi
+
+            echo "NOTE: Creating overlay \"${link}\" for VNIC: ${iface}"
+
+            if ! dladm create-overlay ${overlay_rule} -v "${vxlan_id}" -t "${link}"; then
+                echo "ERROR: Could not create overlay \"${link}\" for VNIC: ${iface}"
+                return 1
+            fi
+        else
+            echo "NOTE: Overlay \"${link}\" already exists for VNIC: ${iface}"
+        fi
+    elif [[ "${tag}" =~ [0-9]$ ]]; then
+        # We have a vnic on etherstub (tag ends with a number)
+        link="${tag}"
+
+        if [[ "${iface}" =~ ^${link}[0-9]+$ ]]; then
+            # Normalize interface name (bar010 -> bar10, foo03 -> foo3)
+            vnic_prefix="$(echo "${link}" | sed 's/\([a-zA-Z_\.-]\)0$/\1/')"
+            vnic_name="${vnic_prefix}${iface:${#link}}"
+        fi
+    else
+        # We have a vnic on normal/aggr interface
+        eval "link=\${SYSINFO_NIC_${tag}}"
+
+        if [[ -z "${link}" ]]; then
+            echo "WARNING: No link found for VNIC: ${iface}"
+            return 0
+        fi
+    fi
+
+    # This is the only difference between start and refresh
+    if [[ "${MODE}" == "refresh" ]] && dladm show-vnic "${vnic_name}" > /dev/null 2>&1; then
+        echo "WARNING: VNIC \"${vnic_name}\" already exists"
+        return 0
+    fi
+
+    # Get vnic_ip parameters
+    eval "vlan=\${${config_source}_${iface}_vlan_id}"
+    eval "macaddr=\${${config_source}_${iface}_mac}"
+    eval "mtu=\${${config_source}_${iface}_mtu}"
+
+    # Call vnic_up
+    if [[ ${ip_version} == "6" ]]; then
+        eval "ip=\${${config_source}_${iface}_ip6}"
+        netmask=""
+
+        echo vnic_up6 "${link}" "${vnic_name}" "${ip}" "${vlan}" "${macaddr}" "${mtu}"
+        vnic_up6 "${link}" "${vnic_name}" "${ip}" "${vlan}" "${macaddr}" "${mtu}"
+    else
+        eval "ip=\${${config_source}_${iface}_ip}"
+        eval "netmask=\${${config_source}_${iface}_netmask}"
+
+        echo vnic_up "${link}" "${vnic_name}" "${ip}" "${netmask}" "${vlan}" "${macaddr}" "" "${mtu}"
+        vnic_up "${link}" "${vnic_name}" "${ip}" "${netmask}" "${vlan}" "${macaddr}" "" "${mtu}"
+    fi
+}
+
+
+#
+# Create overlay_rules.json
+#
+function create_overlay_rules
+{
+    set -o xtrace
+
+    # Create overlay_rules.json
+    config_overlay_rules="$(sdc_config_keys | grep "^overlay_rule_[a-zA-Z0-9_]*[a-zA-Z_]$" | cut -d "_" -f 3-)"
+
+    if [[ -n "${config_overlay_rules}" ]]; then
+        # The overlay configuration folder is placed on swap (does not exist after boot)
```

Comment:
Actually, it's placed on a ramdisk. `Swap` word is misleading even that the statement is true.